openenclave/openenclave
# [CI/CD] Enable RHEL 8 attestation

* Bump Intel SGX driver URLs for Ubuntu. This makes sure that we use the same version on both Ubuntu / RHEL in our Jenkins CI/CD machines.
* Enable attestation for RHEL 8:
  * Remove `flc_enabled=false` when building RHEL 8 Azure image.
  * Do no fail if `flc_enabled` is enabled on RHEL 8.
  * Create symlink for `libdcap_quoteprov.so` into `/usr/lib64`.
  * Add Intel SGX DCAP driver setup on RHEL 8.
  * Install Azure-DCAP-Client from stable branches on RHEL 8.
* Re-enable RHEL 8 testing stages. The linked issues are already solved.
* Refactor pipelines:
  * Fix spacing.
  * Remove code duplication for stages used with / without `FULL_TEST_SUITE`.
* Add RHEL 8 testing with HAS_QUOTE_PROVIDER=ON:
  * This is a validation that attestation is working as expected.
* Integrate RHEL8Test function into ACCTest
  * This commit also fixes space formating for ACCHostVerificationTest function.
* Cleanup Agnostic/Linux/Jenkinsfile
  * Rename `simulationTest` to `simulationContainerTest` since it runs testing into a container.
  * Change `simulationContainerTest` to accept `extra_cmake_args` instead of individual parameters for the cmake flags.

Signed-off-by: Ionut Balutoiu <ibalutoiu@cloudbasesolutions.com>
